tao_compiler/mlir/disc/tests/regression/empty_tensor.cc

```diff
@@ -45,4 +45,18 @@ TEST(EmptyTensor, ImplicitBroadcast2DF32) {
       /*output_descriptors*/ {"f32_X"}));
 }
 
+// Corner case
+// Empty reshape + matmul
+TEST(EmptyTensor, EmptyReshapeMatmul) {
+  EXPECT_TRUE(feature_test_main(
+      /*mlir_file_path*/ c_ft_path + "empty_tensor_reshape_matmul.mlir",
+      /*backend_types*/
+      kSupportedBackendList,
+      /*num_inputs*/ 3,
+      /*num_outputs*/ 1,
+      /*input_descriptors*/ {"0x2x3xf32_X", "6x11xf32_X", "2xi32_h"},
```

Comment:
In which case a dim size could be 0?

Reply:
Yes, empty tensor is valid. Numpy (TF or Pytorch) supports empty tensor. For example,  we may use empty tensor as init value and expand the tensor (e.g. use concat op) by one in each iteration when doing decoding.